Cannot set invoker to public for HTTP request function

According to the typing of `HttpsOptions` class, I should be able to set `invoker` to `public` or `private`. But when I tried to deploy a `https_fn.on_request` function with `invoker=True`, an error is raised:

```
[2024-03-06 02:26:16,162] ERROR in app: Exception on /__/functions.yaml [GET]
Traceback (most recent call last):
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/flask/app.py", line 1463, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/flask/app.py", line 872, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/flask/app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/flask/app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/firebase_functions/private/serving.py", line 122, in get_functions_yaml
    functions = get_functions()
                ^^^^^^^^^^^^^^^
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/firebase_functions/private/serving.py", line 40, in get_functions
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Volumes/Cased/Projects/cloud-functions/functions/main.py", line 29, in <module>
    @https_fn.on_request(invoker="public", memory=options.MemoryOption.MB_512)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/firebase_functions/https_fn.py", line 443, in on_request_inner_decorator
    options._endpoint(func_name=func.__name__),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/Cased/Projects/cloud-functions/functions/.venv/lib/python3.11/site-packages/firebase_functions/options.py", line 1120, in _endpoint
    assert len(
           ^^^^
AssertionError: HttpsOptions: Invalid option for invoker - must be a non-empty list.
```

--- a/src/firebase_functions/options.py
+++ b/src/firebase_functions/options.py
@@ -321,7 +321,8 @@ class RuntimeOptions:
         merged_options: dict = {**global_options, **provider_options}
 
         if self.labels is not None and _GLOBAL_OPTIONS.labels is not None:
-            merged_options["labels"] = {**_GLOBAL_OPTIONS.labels, **self.labels}
+            merged_options["labels"] = {
+                **_GLOBAL_OPTIONS.labels, **self.labels}
         if "labels" not in merged_options:
             merged_options["labels"] = {}
         preserve_external_changes: bool = merged_options.get(
@@ -464,7 +465,7 @@ class TaskQueueOptions(RuntimeOptions):
                 _manifest.TaskQueueTrigger(
                     rateLimits=rate_limits,
                     retryConfig=retry_config,
-                ),
+            ),
         }
         return _manifest.ManifestEndpoint(
             **_typing.cast(_typing.Dict, kwargs_merged))
@@ -853,7 +854,7 @@ class ScheduleOptions(RuntimeOptions):
                     schedule=self.schedule,
                     timeZone=time_zone,
                     retryConfig=retry_config,
-                ),
+            ),
         }
         return _manifest.ManifestEndpoint(
             **_typing.cast(_typing.Dict, kwargs_merged))
@@ -1134,7 +1135,7 @@ class HttpsOptions(RuntimeOptions):
                     invoker = [invoker]
                 assert len(
                     invoker
-                ) > 1, "HttpsOptions: Invalid option for invoker - must be a non-empty list."
+                ) >= 1, "HttpsOptions: Invalid option for invoker - must be a non-empty list."
                 assert "" not in invoker, (
                     "HttpsOptions: Invalid option for invoker - must be a non-empty string."
                 )

--- a/src/firebase_functions/options.py
+++ b/src/firebase_functions/options.py
@@ -321,8 +321,7 @@ class RuntimeOptions:
         merged_options: dict = {**global_options, **provider_options}
 
         if self.labels is not None and _GLOBAL_OPTIONS.labels is not None:
-            merged_options["labels"] = {
-                **_GLOBAL_OPTIONS.labels, **self.labels}
+            merged_options["labels"] = {**_GLOBAL_OPTIONS.labels, **self.labels}
         if "labels" not in merged_options:
             merged_options["labels"] = {}
         preserve_external_changes: bool = merged_options.get(
@@ -465,7 +464,7 @@ class TaskQueueOptions(RuntimeOptions):
                 _manifest.TaskQueueTrigger(
                     rateLimits=rate_limits,
                     retryConfig=retry_config,
-            ),
+                ),
         }
         return _manifest.ManifestEndpoint(
             **_typing.cast(_typing.Dict, kwargs_merged))
@@ -854,7 +853,7 @@ class ScheduleOptions(RuntimeOptions):
                     schedule=self.schedule,
                     timeZone=time_zone,
                     retryConfig=retry_config,
-            ),
+                ),
         }
         return _manifest.ManifestEndpoint(
             **_typing.cast(_typing.Dict, kwargs_merged))

--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -17,6 +17,7 @@ Options unit tests.
 from firebase_functions import options, https_fn
 from firebase_functions import params
 from firebase_functions.private.serving import functions_as_yaml, merge_required_apis
+from pytest import raises
 # pylint: disable=protected-access
 
 
@@ -224,7 +225,5 @@ def test_invoker_with_one_element_doesnt_throw():
 
 
 def test_invoker_with_no_element_throws():
-    try:
+    with raises(AssertionError, match="HttpsOptions: Invalid option for invoker - must be a non-empty list."):
         options.HttpsOptions(invoker=[])._endpoint(func_name="test")
-    except AssertionError:
-        return

--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -225,5 +225,9 @@ def test_invoker_with_one_element_doesnt_throw():
 
 
 def test_invoker_with_no_element_throws():
-    with raises(AssertionError, match="HttpsOptions: Invalid option for invoker - must be a non-empty list."):
+    with raises(
+            AssertionError,
+            match=
+            "HttpsOptions: Invalid option for invoker - must be a non-empty list."
+    ):
         options.HttpsOptions(invoker=[])._endpoint(func_name="test")

--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -221,3 +221,10 @@ def test_merge_apis_duplicate_apis():
 
 def test_invoker_with_one_element_doesnt_throw():
     options.HttpsOptions(invoker=["public"])._endpoint(func_name="test")
+
+
+def test_invoker_with_no_element_throws():
+    try:
+        options.HttpsOptions(invoker=[])._endpoint(func_name="test")
+    except AssertionError:
+        return

--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -44,13 +44,14 @@ def test_global_options_merged_with_provider_options():
     Testing a global option is used when no provider option is set.
     """
     options.set_global_options(max_instances=66)
-    pubsub_options = options.PubSubOptions(topic="foo")  #pylint: disable=unexpected-keyword-arg
+    pubsub_options = options.PubSubOptions(
+        topic="foo")  # pylint: disable=unexpected-keyword-arg
     pubsub_options_dict = pubsub_options._asdict_with_global_options()
     assert (pubsub_options_dict["topic"] == "foo"
-           ), "'topic' property missing from dict"
+            ), "'topic' property missing from dict"
     assert "options" not in pubsub_options_dict, "'options' key should not exist in dict"
     assert (pubsub_options_dict["max_instances"] == 66
-           ), "provider option did not update using the global option"
+            ), "provider option did not update using the global option"
 
 
 def test_https_options_removes_cors():
@@ -119,7 +120,8 @@ def test_merge_apis_empty_input():
     expected_output = []
     merged_apis = merge_required_apis(required_apis)
 
-    assert merged_apis == expected_output, f"Expected {expected_output}, but got {merged_apis}"
+    assert merged_apis == expected_output, f"Expected {
+        expected_output}, but got {merged_apis}"
 
 
 def test_merge_apis_no_duplicate_apis():
@@ -162,7 +164,8 @@ def test_merge_apis_no_duplicate_apis():
 
     merged_apis = merge_required_apis(required_apis)
 
-    assert merged_apis == expected_output, f"Expected {expected_output}, but got {merged_apis}"
+    assert merged_apis == expected_output, f"Expected {
+        expected_output}, but got {merged_apis}"
 
 
 def test_merge_apis_duplicate_apis():
@@ -212,8 +215,12 @@ def test_merge_apis_duplicate_apis():
 
     for expected_item in expected_output:
         assert (expected_item in merged_apis
-               ), f"Expected item {expected_item} missing from the merged list"
+                ), f"Expected item {expected_item} missing from the merged list"
 
     for actual_item in merged_apis:
         assert (actual_item in expected_output
-               ), f"Unexpected item {actual_item} found in the merged list"
+                ), f"Unexpected item {actual_item} found in the merged list"
+
+
+def test_invoker_with_one_element_doesnt_throw():
+    options.HttpsOptions(invoker=["public"])._endpoint(func_name="test")

--- a/tests/test_options.py
+++ b/tests/test_options.py
@@ -44,14 +44,13 @@ def test_global_options_merged_with_provider_options():
     Testing a global option is used when no provider option is set.
     """
     options.set_global_options(max_instances=66)
-    pubsub_options = options.PubSubOptions(
-        topic="foo")  # pylint: disable=unexpected-keyword-arg
+    pubsub_options = options.PubSubOptions(topic="foo")  # pylint: disable=unexpected-keyword-arg
     pubsub_options_dict = pubsub_options._asdict_with_global_options()
     assert (pubsub_options_dict["topic"] == "foo"
-            ), "'topic' property missing from dict"
+           ), "'topic' property missing from dict"
     assert "options" not in pubsub_options_dict, "'options' key should not exist in dict"
     assert (pubsub_options_dict["max_instances"] == 66
-            ), "provider option did not update using the global option"
+           ), "provider option did not update using the global option"
 
 
 def test_https_options_removes_cors():
@@ -120,8 +119,7 @@ def test_merge_apis_empty_input():
     expected_output = []
     merged_apis = merge_required_apis(required_apis)
 
-    assert merged_apis == expected_output, f"Expected {
-        expected_output}, but got {merged_apis}"
+    assert merged_apis == expected_output, f"Expected {expected_output}, but got {merged_apis}"
 
 
 def test_merge_apis_no_duplicate_apis():
@@ -164,8 +162,7 @@ def test_merge_apis_no_duplicate_apis():
 
     merged_apis = merge_required_apis(required_apis)
 
-    assert merged_apis == expected_output, f"Expected {
-        expected_output}, but got {merged_apis}"
+    assert merged_apis == expected_output, f"Expected {expected_output}, but got {merged_apis}"
 
 
 def test_merge_apis_duplicate_apis():
@@ -173,7 +170,7 @@ def test_merge_apis_duplicate_apis():
     This test evaluates the merge_required_apis function when the
     input list contains duplicate APIs with different reasons.
     The desired outcome for this test is a list where the duplicate
-    APIs are merged properly and reasons are combined. 
+    APIs are merged properly and reasons are combined.
     This test ensures that the function correctly merges the duplicate
     APIs and combines the reasons associated with them.
     """
@@ -215,11 +212,11 @@ def test_merge_apis_duplicate_apis():
 
     for expected_item in expected_output:
         assert (expected_item in merged_apis
-                ), f"Expected item {expected_item} missing from the merged list"
+               ), f"Expected item {expected_item} missing from the merged list"
 
     for actual_item in merged_apis:
         assert (actual_item in expected_output
-                ), f"Unexpected item {actual_item} found in the merged list"
+               ), f"Unexpected item {actual_item} found in the merged list"
 
 
 def test_invoker_with_one_element_doesnt_throw():

